lib/seattleflu/db/cli/command/etl/longitudinal.py

```diff
@@ -0,0 +1,356 @@
+
+
+def age(document: dict) -> str:
+    """
+    Given a *document*, retrieve age value and
+    return as a string to fit the interval format.
+
+    If no value is given for age, then will just return None.
+    """
+    age = document.get("age")
+    if age is None:
+        return None
+    return f"{float(age)} years"
+
+
+def encounter_details(document: dict) -> dict:
+    """
+    Describe encounter details in a simple data structure designed to be used
+    from SQL.
+    """
+    return {
+            "age": age_to_delete(document.get("age")), # XXX TODO: Remove age from details
+            "locations": {
+                "home": {
+                    "region": document.get("census_tract"),
+                }
+            },
+            "responses": {
+                "Race": race(document.get("Race")),
+                "FluShot": flu_shot(document.get("FluShot")),
+                "AssignedSex": [sex(document.get("AssignedSex"))],
+                "HispanicLatino": hispanic_latino(document.get("HispanicLatino")),
+                "MedicalInsurance": insurance(document.get("MedicalInsurance"))
+            },
+        }
+
+def age_to_delete(age: float) -> str:
+    """
+    TODO: Delete this function once we remove age from details
+    Given an *age*, return a dict containing its 'value' and a boolean for
+    'ninetyOrAbove'.
+    Currently applys math.ceil() to age to match the age from Audere.
+    This may change in the future as we push to report age in months for
+    participants less than 1 year old.
+    If no value is given for *age*, then will just retun None.
+    """
+    if age is None:
+        return None
+
+    return {
+        "value": min(ceil(age), 90),
+        "ninetyOrAbove": ceil(age) >= 90
+    }
+
+
+def race(races: list) -> list:
+    """
+    Given a *race_name*, returns the matching race identifier found in Audere
+    survey data. Converts multiple races to 'other'.
+    """
+    if races is None:
+        LOG.debug("No race response found.")
+        return [None]
+
+    if len(races) > 1:
+        return 'other'
+
+    race_map = {
+        "native_american": "americanIndianOrAlaskaNative",
+        "asian": "asian",
+        "black": "blackOrAfricanAmerican",
+        "native_hawaiian": "nativeHawaiian",
+        "white": "white",
+        "other": "other",
+        "unknown": "unknown",  # TODO is this what we want, or None?
+    }
+
+    race_name = races[0]
+    if race_name not in race_map:
+        raise UnknownRaceError(f"Unknown race name «{race_name}»")
+
+    return [race_map[race_name]]
+
+
+def hispanic_latino(ethnic_group: str) -> list:
+    """
+    Given an *ethnic_group*, returns yes/no value for HispanicLatino key.
+    """
+    if ethnic_group is None:
+        LOG.debug("No ethnic group response found.")
+        return [None]
+
+    ethnic_map = {
+        1: "yes",
+        2: "no",
+        3: "refused",  # TODO: is this what we want, or None instead?
+    }
```

Comment:
is there precedence for a value other than yes/no here?

Reply:
I think it's either yes/no or null